src/components/tools/geometry-tool/movable-line-dialog.tsx

```diff
@@ -1,148 +1,28 @@
-        </div>
-      </Dialog>
-    );
-  }
-
-  private handleSlopeChange = (evt: React.FormEvent<HTMLInputElement>) => {
-    this.setState({slope: (evt.target as HTMLInputElement).value });
-  };
-
-  private handleInterceptChange = (evt: React.FormEvent<HTMLInputElement>) => {
-    this.setState({intercept: (evt.target as HTMLInputElement).value });
-  };
-
-  private getValidationError = () => {
-    const { line } = this.props;
-    const slope = parseFloat(this.state.slope);
-    const intercept = parseFloat(this.state.intercept);
-    if (!isFinite(slope) || !isFinite(intercept)) {
-      return "Please enter a valid number for slope and intercept.";
-    }
-    const intersections = getBoundingBoxIntersections(slope, intercept, line.board);
-    if (intersections.length < 2) {
-      return "Please change the graph axes scale to make the line visible, and try again.";
-    }
-  };
-
-  private getLineControlPoints = () => {
-    const { line } = this.props;
-    const slope = parseFloat(this.state.slope);
-    const intercept = parseFloat(this.state.intercept);
-    if (isFinite(slope) && isFinite(intercept)) {
-      const board = line.board;
-      const intersections = getBoundingBoxIntersections(slope, intercept, board);
-      if (!intersections.length) return undefined;
-      let point1: [number, number];
-      let point2: [number, number];
-      if (board.hasPoint(0, intercept)) {
-        point1 = [0, intercept];
-        const maxX = intersections[1][0];
-        const middleX = maxX / 2;
-        point2 = [middleX, solveForY(slope, intercept, middleX)];
-      } else {
-        const minX = intersections[0][0];
-        const maxX = intersections[1][0];
-        const lineWidth = maxX - minX;
-        const thirdLineWidth = lineWidth / 3;
-        const x1 = minX + thirdLineWidth;
-        const x2 = minX + thirdLineWidth * 2;
-        point1 = [x1, solveForY(slope, intercept, x1)];
-        point2 = [x2, solveForY(slope, intercept, x2)];
-      }
-      return [point1, point2];
-    } else {
-      return undefined;
-    }
-  };
-
-  private handleAccept = () => {
-    const { line, onAccept } = this.props;
-    const points = this.getLineControlPoints();
-    if (points) {
-      onAccept(line, points[0], points[1]);
-    } else {
-      this.handleCancel();
-    }
-  };
-
-  private handleCancel = () => {
-    if (this.props.onClose) {
-      this.props.onClose();
-    }
-  };
-
-  private handleKeyDown = (evt: React.KeyboardEvent<HTMLInputElement>) => {
-    evt.stopPropagation();
-    if (evt.keyCode === 13) {
-      this.handleAccept();
-    } else if (evt.keyCode === 27) {
-      this.handleCancel();
-    }
-  };
-
-}
+// Component wrapper for useMovableLineDialog() for use by class components.
+const MovableLineDialog: React.FC<IProps> = ({
+  line, onAccept, onClose
+}) => {
+
+  const [showDialog, hideDialog] = useMovableLineDialog({
+    line,
+    onAccept,
+    onClose
+  });
```

Comment:
A nit, but in cases like these where the code could easily fit on one line I'd be inclined to do so.